app/assets/javascripts/src/Metamaps.Map.js

```diff
@@ -436,11 +436,25 @@ Metamaps.Map.InfoBox = {
     })
     $('body').click(self.close)
 
-    self.attachEventListeners()
-
-    
+    self.attachEventListeners() 
 
     self.generateBoxHTML = Hogan.compile($('#mapInfoBoxTemplate').html())
+    
+    var querystring = window.location.search.replace(/^\?/, '')
+    if (querystring == 'new') { 
+      self.open()
+      window.setTimeout(self.requestMapTitle, 3000)
+    }
+  },
+  requestMapTitle: function () {
+    var t = prompt('Please enter a name for your map. You can also set it any time using the map info panel.')
```

Comment:
The prompt IS pretty 90s. There are a few options here and I'm open to them:

Just open the map info box, and draw attention somehow to the Map Name

Do a little custom prompt that is nicer

On Aug 15, 2016 8:43 PM, "Devin Howard" <notifications@github.com> wrote:

> In app/assets/javascripts/src/Metamaps.Map.js
> <https://github.com/metamaps/metamaps/pull/607#discussion_r74860912>:
>
> >
> >      self.generateBoxHTML = Hogan.compile($('#mapInfoBoxTemplate').html())
> > +
> > +    var querystring = window.location.search.replace(/^\?/, '')
> > +    if (querystring == 'new') {
> > +      self.open()
> > +      window.setTimeout(self.requestMapTitle, 3000)
> > +    }
> > +  },
> > +  requestMapTitle: function () {
> > +    var t = prompt('Please enter a name for your map. You can also set it any time using the map info panel.')
>
> Are prompts too 90's? Should we get some kind of modal dialog library,
> like bootbox or something?
>
> I'm thinking this sentence could be better..... not exactly sure. Maybe
>
> Please enter a name for your map. You can change this later at any time
> using the map info box.
>
> Box/panel? Is it clear what that means? Before the map info box could be
> blissfully ignored if you were new to Metamaps, but now it's a new
> introduced concept. Dunno. How do people do those interactive tutorials on
> people's first sign ins?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/metamaps/metamaps/pull/607/files/c17fee44494264a7877bab6a458517e9bae456e0#r74860912>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABWAYRhnqKErJ3l4oiVjQlyRXpF67RiHks5qgQelgaJpZM4Jkugb>
> .
>